### PR TITLE
support add feature in "alter replica_query rule"

### LINKED ISSUE #8979: support `add feature` in alter `replica_query rule`
## Feature Request
support `add feature` in alter `replica_query rule`

### Describe the feature you would like.
maybe grammar like this:

```
// alter for add rule
ALTER REPLICA_QUERY RULE  ADD RULE (logic_ds_a (primary=primary_new_ds_a, repilca=repilca_ds_a_0) RANDOM(xxx=xxx),...)

// alter for modifiy exist rule
ALTER REPLICA_QUERY RULE  (logic_ds_a (primary=primary_new_ds_a, repilca=repilca_ds_a_0) RANDOM(xxx=xxx),...)
```

### PR DESCRIPTION
**Pull request by jingshanglu**
Fixes #8979 .

### CODE REVIEW

File: shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java [hunk 1]
```diff
@@ -37,6 +37,12 @@
     
     RESOURCE_NOT_EXIST(11004, "C11004", "Can not drop resources, resources %s do not exist."),
     
+    REPLICA_QUERY_RULE_NOT_EXIST(11005, "C11005", "Replica query rule do not exist."),
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java:40]: `do not` should be `does not`

File: shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java [hunk 2]
```diff
@@ -37,6 +37,12 @@
...
+    
+    REPLICA_QUERY_RULE_DATASOURCE_NOT_EXIST(11006, "C11006", "Datasource %s in Replica query rule do not exist."),
```
Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java:42]: data source are 2 words

Comment (terrymanu) [inline shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java:42]: `Replica` should be lower case if not in the first word of sentence.

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by terrymanu** [comment]
Please add label and milestone

File: shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java (hunk 1 above)
Comment (jingshanglu) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java]: ok

File: shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java (hunk 2 above)
Comment (jingshanglu) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java]: ok

Comment (jingshanglu) [reply shardingsphere-db-protocol/shardingsphere-db-protocol-core/src/main/java/org/apache/shardingsphere/db/protocol/error/CommonErrorCode.java]: ok

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/impl/AlterReplicaQueryRuleBackendHandler.java [hunk 3]
```diff
@@ -49,34 +57,73 @@
...
+            resources.addAll(each.getReplicaDatasources());
+        }
+        Map<String, DataSource> datasourceMap = ProxyContext.getInstance().getMetaData(schemaName).getResource().getDataSources();
+        Collection<String> notExistResource = resources.stream().filter(each -> !datasourceMap.containsKey(each)).collect(Collectors.toList());
+        if (!notExistResource.isEmpty()) {
+            throw new ResourceNotExistedException(notExistResource);
+        }
+    }
+    
+    private void checkModifyDatasourceNotExist(final AlterReplicaQueryRuleStatement statement, final String schemaName) {
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/impl/AlterReplicaQueryRuleBackendHandler.java:114]: Data source are 2 words

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/impl/AlterReplicaQueryRuleBackendHandler.java (hunk 3 above)
Comment (jingshanglu) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/impl/AlterReplicaQueryRuleBackendHandler.java]: ok

**Review by terrymanu — APPROVED** [review]

**Pull request merged**